### PR TITLE
TE-1362 Extend `Reviews` with submit form

### PR DESCRIPTION
**Pull request by red-runner**
[Related YouTrack issue](https://youtrack.lodgify.net/issue/TE-1362)

### What **one** thing does this PR do?
* Implements a submit form behaviour for the `Reviews` component; described in this [YouTrack issue](https://youtrack.lodgify.net/issue/TE-1339)

### Any other notes
* It also upgrades the `RatingInput` component exposing a label prop

#### Default

standard reviews

#### Content

content_strings

#### Usage Validation

usage_validation

#### State Success

state_success

#### State Error

state_error

### CODE REVIEW

File: src/components/property-page-widgets/Reviews/Readme.md [hunk 1]
```diff
@@ -2,21 +2,35 @@
...
     reviewerCategory: 'Young people',
     reviewerLocation: 'Lithuania',
     reviewerName: 'Magellan',
     reviewerStayDate: '9/2015',
-    reviewResponse: {
-      dateTime: '10/14/2015 12:22:58 PM',
-      source: 'The Owner',
-      text:
-        'You can also personally respond to each review: Thanks for you kind review, James! Hope to welcome you back soon!',
-    },
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Reviews/Readme.md]: Retain the `reviewResponse` for this first basic example, but exclude it from the others to reduce noise.

File: src/components/inputs/RatingInput/component.js [hunk 2]
```diff
@@ -1,35 +1,41 @@
...
+  </Fragment>
 );
 
 Component.displayName = 'RatingInput';
 
 Component.defaultProps = {
   error: false,
   iconSize: 'small',
-  name: '',
+  label: RATINGINPUT_TEXT,
```
Comment (marcdavi-es) [inline src/components/inputs/RatingInput/component.js]: I think in other inputs, default `label` value is `''` ?

File: src/components/inputs/RatingInput/component.js [hunk 3]
```diff
@@ -38,6 +44,7 @@ Component.propTypes = {
   error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
   /** The size of the star icons. */
   iconSize: PropTypes.oneOf(['tiny', 'small']),
+  label: PropTypes.string,
```
Comment (marcdavi-es) [inline src/components/inputs/RatingInput/component.js]: Needs a prop doc comment.

File: src/components/inputs/RatingInput/constants.js [hunk 4]
```diff
@@ -1,2 +1,3 @@
 export const MAX_RATING = 5;
 export const ONRATE_HANDLER = 'onRate';
```
Comment (marcdavi-es) [inline src/components/inputs/RatingInput/constants.js]: and `RATING_INPUT_TEXT`

File: src/utils/default-strings/constants.js [hunk 5]
```diff
@@ -37,6 +40,7 @@ export const PHONE = 'Phone';
 export const PRICE_PER_EXTRA_PER = 'Price per extra per.';
 export const PROPERTY = 'Property';
 export const PROPERTY_PICTURES = 'Property pictures';
+export const REVIEW_FORM_HEADING = 'Add a Review';
```
Comment (marcdavi-es) [inline src/utils/default-strings/constants.js]: All of these identifiers match the strings. This should be 

```js
export const ADD_A_REVIEW = 'Add a Review';
```

**Comment by red-runner** [comment]
@marcdavi-es  comments are solved

File: src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.js [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+import { RatingInput } from 'inputs/RatingInput';
+
+/**
+ * @param {Object} props
+ * @param {string} props.apartmentInputLabel
+ * @param {string} props.commentInputLabel
+ * @param {string} props.emailInputLabel
+ * @param {string} props.errorMessage
+ * @param {string} props.guestTypeInputLabel
+ * @param {Object[]} props.guestTypeOptions
```
Comment (stephenkr) [inline src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.js]: Can you vertically align these props?

File: src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.js [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+  propertyOptions,
+  ratingAverage,
+  ratingText,
+  reviewFormHeading,
+  submitButtonText,
+  successMessage,
+  titleInputLabel,
+  validation,
+  yearInputLabel,
+  yearsOption,
```
Comment (stephenkr) [inline src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.js]: Don't forget to add in the eslint enable comment after the props.
```
/* eslint-enable react/prop-types */
```

File: src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.spec.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+  reviewFormHeading: 'Beheading',
+  submitButtonText: 'sumbmutbuttuntuxt',
+  successMessage: 'succuss',
+  titleInputLabel: 'Duke',
+  validation,
+  yearInputLabel: 'yuur',
+  yearOptions,
+};
+
+const getModalForm = props => mount(<div>{getModalFormMarkup(props)}</div>);
```
Comment (stephenkr) [inline src/components/property-page-widgets/Reviews/utils/getModalFormMarkup.spec.js]: I don't think you need to wrap the utility with the `div` tags, I think `mount(getModalFormMarkup(props))` is enough.

File: src/components/property-page-widgets/Reviews/component.js [hunk 9]
```diff
@@ -65,17 +78,115 @@ export const Component = ({
...
+      /** The underlying value for the option. */
+      value: PropTypes.oneOfType([
+        PropTypes.bool,
+        PropTypes.number,
+        PropTypes.string,
+      ]),
+    })
+  ),
   /** The text to display as a heading at the top of the widget. */
+  // eslint-disable-next-line react/no-unused-prop-types
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Reviews/component.js]: No need for this.

File: src/components/property-page-widgets/Reviews/component.js [hunk 10]
```diff
@@ -65,17 +78,115 @@ export const Component = ({
...
   /** The average numeral rating for the properties. */
   ratingAverage: PropTypes.number.isRequired,
+  /** A visible label to display with the rating stars. */
+  // eslint-disable-next-line react/no-unused-prop-types
+  ratingText: PropTypes.string,
+  /** The text to display as a heading at the top of the modal form */
+  // eslint-disable-next-line react/no-unused-prop-types
+  reviewFormHeading: PropTypes.string,
   /** The collection of reviews. */
+  // eslint-disable-next-line react/no-unused-prop-types
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Reviews/component.js]: Same as https://github.com/lodgify/lodgify-ui/pull/439/commits/905f33e464d0836a757fdd3b1546e85585924718#r235345386

File: src/components/property-page-widgets/Reviews/component.js [hunk 11]
```diff
@@ -65,17 +78,115 @@ export const Component = ({
...
+        PropTypes.number,
+        PropTypes.string,
+      ]),
+    })
+  ),
   /** The average numeral rating for the properties. */
   ratingAverage: PropTypes.number.isRequired,
+  /** A visible label to display with the rating stars. */
+  // eslint-disable-next-line react/no-unused-prop-types
+  ratingText: PropTypes.string,
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Reviews/component.js]: This should be called `ratingInputLabel`, right?

File: src/components/property-page-widgets/Reviews/component.js [hunk 12]
```diff
@@ -65,17 +78,115 @@ export const Component = ({
...
+  /** The text to display as a heading at the top of the modal form */
+  // eslint-disable-next-line react/no-unused-prop-types
+  reviewFormHeading: PropTypes.string,
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Reviews/component.js]: `reviewFormHeadingText`

File: src/components/property-page-widgets/Reviews/component.js [hunk 13]
```diff
@@ -104,5 +215,42 @@ Component.propTypes = {
...
+    phone: PropTypes.object,
+    property: PropTypes.object,
+    room: PropTypes.object,
+  }),
+  /** The label for the year input. */
+  // eslint-disable-next-line react/no-unused-prop-types
+  yearInputLabel: PropTypes.string,
+  /** The options which the user can select for the year field. */
+  // eslint-disable-next-line react/no-unused-prop-types
+  yearsOption: PropTypes.arrayOf(
```
Comment (marcdavi-es) [inline src/components/property-page-widgets/Reviews/component.js]: `yearsOptions`

File: src/utils/default-strings/constants.js [hunk 14]
```diff
@@ -37,6 +40,7 @@ export const PHONE = 'Phone';
 export const PRICE_PER_EXTRA_PER = 'Price per extra per.';
 export const PROPERTY = 'Property';
 export const PROPERTY_PICTURES = 'Property pictures';
+export const ADD_A_REVIEW = 'Add a Review';
```
Comment (marcdavi-es) [inline src/utils/default-strings/constants.js]: Dude - this whole file is alphabetical for ease of indexing.

File: package.json [hunk 15]
```diff
@@ -20,7 +20,7 @@
   },
   "size-limit": [
     {
-      "limit": "258 KB",
+      "limit": "265 KB",
```
Comment (stephenkr) [inline package.json]: Can you set this to `259 KB`?

Looking at the logs in the CI, you can see the returned bundle size is 258.73 KB.

screen shot 2018-11-22 at 10 43 56

Comment (marcdavi-es) [reply package.json]: @stephenkr Why?

Comment (stephenkr) [reply package.json]: I thought we were trying to keep the bundle size limit as small as possible?

File: src/components/inputs/RatingInput/component.js [hunk 16]
```diff
@@ -1,35 +1,41 @@
...
 );
 
 Component.displayName = 'RatingInput';
 
 Component.defaultProps = {
   error: false,
   iconSize: 'small',
-  name: '',
+  label: '',
+  name: 'ratingInput',
```
Comment (marcdavi-es) [inline src/components/inputs/RatingInput/component.js]: This isn't consistent with any of the other inputs.

File: package.json (hunk 15 above)
Comment (marcdavi-es) [reply package.json]: But we nudge it up every time we exceed the 'limit'.

Right now we don't have a budget in force in practice - so there's not substantive difference between increasing this by 1KB or 6KB.

**Comment by red-runner** [comment]
@marcdavi-es 
* Changed `Readme` for the `RatingInput` to make it more consistent with other inputs.
* Set name `defaultProp` for the `RatingInput` to `''` to make it more consistent with other inputs.
* Changed `Readme` for the `Reviews` to make it more consistent with other components with forms.
* I've made validation object explicit in the `Reviews` `Readme` to make it more explicit. 
* Added `name="rating"` in the `getModalFormMarkup.js` to make it consistent with the rest of the inputs. 
* Solved indentation issues in `Readme` files, wrong spacing etc.

**Pull request merged**